packages/core/strapi/src/Strapi.ts

```diff
@@ -372,18 +372,23 @@ class Strapi implements StrapiI {
     // Emit started event.
     // do not await to avoid slower startup
     // This event is anonymous
-    this.telemetry.send('didStartServer', {
-      groupProperties: {
-        database: this.config.get('database.connection.client'),
-        plugins: Object.keys(this.plugins),
-        numberOfAllContentTypes: _.size(this.contentTypes), // TODO: V5: This event should be renamed numberOfContentTypes in V5 as the name is already taken to describe the number of content types using i18n.
-        numberOfComponents: _.size(this.components),
-        numberOfDynamicZones: getNumberOfDynamicZones(),
-        environment: this.config.environment,
-        // TODO: to add back
-        // providers: this.config.installedProviders,
-      },
-    });
+    this.telemetry
+      .send('didStartServer', {
+        groupProperties: {
+          database: this.config.get('database.connection.client'),
+          plugins: Object.keys(this.plugins),
+          numberOfAllContentTypes: _.size(this.contentTypes), // TODO: V5: This event should be renamed numberOfContentTypes in V5 as the name is already taken to describe the number of content types using i18n.
+          numberOfComponents: _.size(this.components),
+          numberOfDynamicZones: getNumberOfDynamicZones(),
+          numberOfCustomControllers: Object.values<Common.Controller>(this.controllers).filter(
+            factories.isCustomController
+          ).length,
```

Comment:
Note for the future, but not necessarily your issue:

We have a catch on the `send`, but we don't have a catch around the actual block (eg, if `isCustomController`, `getNumberOfDynamicZones`, `config.get`, or anything else were to throw), we should add that so that it's impossible for telemetry to throw an error now that it's getting more complex